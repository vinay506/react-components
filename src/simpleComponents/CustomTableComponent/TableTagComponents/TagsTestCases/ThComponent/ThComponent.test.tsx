import React from 'react';
import { shallow } from 'enzyme';
import { CustomThCell } from '../../index';
import renderer from 'react-test-renderer'


describe('<Custom TH Cell >', function () {
    let wrapper: any;
    let props: any;
    let tree: any;
    beforeEach(() => {
        props = {
            style:{fontWeight:'bold',height:'45px'}
        };
        wrapper = shallow(<CustomThCell {...props}>Sample value for th</CustomThCell>)
    });

    // Snapshot test
    it('should render ', () => {
        tree = renderer.create(<CustomThCell {...props}></CustomThCell>).toJSON()
        expect(tree).toMatchSnapshot()
    })

      it('should render Th cell', () => {
        const th = wrapper.find('th');
        expect(th.exists()).toBe(true);
    })

    it('should match the value', () => {
        const value = wrapper.find('th').text();
        expect(value).toEqual('Sample value for th')
    })

    it('should match with classnames of th', () => {
        const props = wrapper.find('th').props();
        expect(props.className).toEqual('rmwc-data-table__cell mdc-data-table__header-cell')
    })

    it('should match with styles from props', () => {
        const props = wrapper.find('th').props();
        expect(props.style).toEqual({"fontWeight":"bold","height":"45px"});
    })

    
})