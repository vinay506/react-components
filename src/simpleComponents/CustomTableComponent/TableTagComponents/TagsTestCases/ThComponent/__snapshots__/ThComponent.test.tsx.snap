// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom TH Cell > should render  1`] = `
<th
  className="rmwc-data-table__cell mdc-data-table__header-cell"
  style={
    Object {
      "fontWeight": "bold",
      "height": "45px",
    }
  }
/>
`;
